crates/router/src/connector/klarna/transformers.rs

```diff
@@ -252,32 +402,117 @@ impl TryFrom<types::PaymentsResponseRouterData<KlarnaPaymentsResponse>>
-                item.data.request.is_auto_capture()?,
-            )),
-            connector_response: Some(connector_response),
-            ..item.data
-        })
+        let payment_method_data = item.data.request.payment_method_data.clone();
+
+        match payment_method_data{
+            domain::PaymentMethodData::PayLater(domain::PayLaterData::KlarnaSdk { .. }) => {
+                    Ok(Self {
+                        response: Ok(types::PaymentsResponseData::TransactionResponse {
+                            resource_id: types::ResponseId::ConnectorTransactionId(
+                                item.response.order_id.clone(),
+                            ),
+                            redirection_data:Box::new(None),
+                            mandate_reference: Box::new(None),
+                            connector_metadata: None,
+                            network_txn_id: None,
+                            connector_response_reference_id: Some(item.response.order_id.clone()),
+                            incremental_authorization_allowed: None,
+                            charge_id: None,
+                        }),
+                        status: enums::AttemptStatus::foreign_from((
+                            item.response.fraud_status.unwrap_or(KlarnaFraudStatus::Accepted),
+                            item.data.request.is_auto_capture()?,
+                        )),
+                        connector_response: Some(connector_response),
+                        ..item.data
+                    })
+                },
+                domain::PaymentMethodData::PayLater(domain::PayLaterData::KlarnaCheckout {})  => {
+
+                    Ok(Self {
+                        response: Ok(types::PaymentsResponseData::TransactionResponse {
+                            resource_id: types::ResponseId::ConnectorTransactionId(
+                                            item.response.order_id.clone(),
+                                        ),
+                            redirection_data: Box::new(Some(RedirectForm::KlarnaCheckout {
+                                html_snippet: item.response.html_snippet.clone().unwrap_or_default(),
+                            })),
+                            mandate_reference: Box::new(None),
+                            connector_metadata: None,
+                            network_txn_id: None,
+                            connector_response_reference_id: None,
+                            incremental_authorization_allowed: None,
+                            charge_id: None,
+                        }),
+                        status: enums::AttemptStatus::from(item.response.status.unwrap_or(KlarnaCheckoutStatus::CheckoutComplete)),
+                        connector_response: Some(connector_response),
+                        ..item.data
+                    })
+                },
+            hyperswitch_domain_models::payment_method_data::PaymentMethodData::Card(_card) => {
```

Comment:
combine the cases with the same outcome using the | operator

Reply:
fixed